src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,117 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int INITIAL_CAPACITY = 10;
+    private int size;
+    private T[] elementsData;
+
+    public ArrayList() {
+        elementsData = (T[]) new Object[INITIAL_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == elementsData.length) {
+            resize();
+        }
+        elementsData[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        checkIndex(index);
+        if (index == size) {
+            add(value);
+        } else {
+            if (size == elementsData.length) {
+                resize();
+            }
+            System.arraycopy(elementsData, index, elementsData, index + 1, size - index);
+            elementsData[index] = value;
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        T[] data = (T[]) new Object[list.size()];
+        for (int i = 0; i < list.size(); i++) {
+            data[i] = list.get(i);
+        }
+        if (elementsData.length - size < list.size()) {
+            int newSize = ((int) (elementsData.length * 1.5 + 1) + list.size());
```

Comment:
this is code duplication, because we have similar code in method resize().
1 the simplest solution is to make method `addAll()` more concise and simple call the method `add(value)` in the loop
2 the second variant is add parameter to method `resize() ` which will resize not only by multiplying  by `1.5`, but adding needed quantity of elements (if we know that quantity of elements in more the `0.5 * size`)